src/ServicePulse.Host/vue/src/components/failedmessages/DeletedMessageGroups.vue

```diff
@@ -1,60 +1,79 @@
-<script setup>
+<script setup lang="ts">
 import { onMounted, onUnmounted, ref } from "vue";
 import { useRoute, useRouter } from "vue-router";
 import { licenseStatus } from "../../composables/serviceLicense";
 import { connectionState, stats } from "../../composables/serviceServiceControl";
 import { useShowToast } from "../../composables/toast";
 import { useGetArchiveGroups, useRestoreGroup } from "../../composables/serviceMessageGroup";
-import { useFetchFromServiceControl } from "../../composables/serviceServiceControlUrls";
+import { useTypedFetchFromServiceControl } from "../../composables/serviceServiceControlUrls";
 import { useCookies } from "vue3-cookies";
 import NoData from "../NoData.vue";
 import TimeSince from "../TimeSince.vue";
 import LicenseExpired from "../../components/LicenseExpired.vue";
 import ServiceControlNotAvailable from "../ServiceControlNotAvailable.vue";
 import ConfirmDialog from "../ConfirmDialog.vue";
 import routeLinks from "@/router/routeLinks";
+import FailureGroupView from "@/resources/FailureGroupView";
+import { TYPE } from "vue-toastification";
+
+interface WorkflowState {
+  status: string;
```

Comment:
Should this be typed? It looks like there are a specific set of allowed strings.

Reply:
https://github.com/Particular/ServicePulse/pull/1780/files